### PR TITLE
UI framework continued - weight-defined layout, ObjectsTablePanel

### LINKED ISSUE #6134: UI Components: Base level template renderer
Base level component that renders an entire template (still as a single panel).

template_name and panel_title as input.

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes #6134
# What's Changed

- As discussed for a design refinement, remove the `Layout` class in favor of 1) letting a `Tab` choose from the new `LayoutChoices` ChoiceSet and 2) letting each `Panel` provide a `weight` and a `section` from the new `SectionChoices` ChoiceSet
- Tabs are now automatically ordered by `weight`, as are Panels within a section.
    - Future: figure out a plan for App `TemplateExtension` functionality - could/should we allow them to directly define Tabs and Panels instead?
- Add `ObjectsTablePanel` and use it for the Contact Associations, Object Metadata, and Dynamic Groups tables.

# Screenshots

image

image

image

image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/ui/object_detail.py [hunk 1]
```diff
@@ -219,47 +299,73 @@ class _ObjectDetailAdvancedTab(Tab):
...
-                        ObjectFieldsPanel(
-                            label="Data Provenance",
-                            fields=["created", "last_updated"],
-                            ignore_nonexistent_fields=True,
-                            # TODO add created_by/last_updated_by derived fields and link to REST API
-                        ),
-                        # TODO add advanced_ui custom fields and relationships
-                    ),
+    def __init__(self, **kwargs):
+        kwargs.setdefault("weight", 1000)
```
Comment (HanlinMiao) [inline nautobot/core/ui/object_detail.py:303]: Doesn't advanced tab always come second?

File: nautobot/core/ui/object_detail.py [hunk 2]
```diff
@@ -219,47 +299,73 @@ class _ObjectDetailAdvancedTab(Tab):
...
-    def __init__(self):
-        super().__init__(
-            layouts=(
-                LayoutFullWidth(
-                    panels=(
-                        # TODO
-                        TemplatePanel(label="Contact Associations", template_string="Contact Associations Table"),
-                    ),
+    def __init__(self, **kwargs):
+        kwargs.setdefault("weight", 1100)
```
Comment (HanlinMiao) [inline nautobot/core/ui/object_detail.py:334]: I am a little confused as to why we are starting from 0 in the ObjectDetailMainTab and jump to 1000 in the ObjectDetailAdvancedTab.

File: nautobot/core/ui/object_detail.py [hunk 3]
```diff
@@ -219,47 +299,73 @@ class _ObjectDetailAdvancedTab(Tab):
...
+            <span class="mdi mdi-trash-can-outline" aria-hidden="true"></span> Delete
+        </button>
+    {% endif %}
+        {% if perms.extras.add_contactassociation %}
+            <div class="pull-right">
+            <a href="{% url 'extras:object_contact_team_assign' %}?return_url={{return_url}}&associated_object_id={{object.id}}&associated_object_type={{content_type.id}}" class="btn btn-primary btn-xs">
+                <span class="mdi mdi-plus-thick" aria-hidden="true"></span> Add Contact
+            </a>
+        </div>
+    {% endif %}
```
Comment (HanlinMiao) [inline nautobot/core/ui/object_detail.py:361]: This might be GitHub rendering but is this block indented correctly?

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/core/ui/object_detail.py (hunk 1 above)
Comment (glennmatthews) [reply nautobot/core/ui/object_detail.py:303]: Probably it should, good call.

**Pull request merged**